Fix druid-jdbc issues

metamben opened the pull request:
Fixes #42237.

kamilmielnik committed:
FE - Temporarily mock metrics CRUD endpoints (#38357)

* shortcut to introduce report_card.type. WIP

* major test fixes

* test fixes

* docs and test fixes

* revision for type

* Add type property to Card

* Add type property to CardCompletionItem

* Revert "Add type property to CardCompletionItem"

This reverts commit 3c2db2aa2fbdce05d676711ae665ee4c5c73037a.

* Use "type" attribute to create a model in tests

* test fixes and some of Chris's suggestions

* test fixes and add revision migration

* we shouldn't dissoc dataset on migration

* warn if type not found when trying to ensure type and datset consistency

* default value for type when create card

* no need to comment out in tests

* Use "type" attribute to create a model in tests

* Refactor cy.createNativeQuestion & cy.createQuestion to use "type" instead "dataset" option

* Rename type to queryType

* Rename entityType to type

* Improve JSDoc

* Consolidate conditions

* Fix missing "type" attribute

* Revert whitespace change

* Fix typing

* om well, what's :or?

* okay makes the :dataset works

* Introduce Question.prototype.setType

* Use Question.prototype.setType instead of Question.prototype.setDataset

* Deprecate Question.prototype.setType

* Inline Question.prototype.type

* Introduce new Question.prototype.type

* Type _setDataset

* Use Question.prototype.type instead of Question.prototype.isDataset

* Remove Question.prototype.isDataset

* Use nullish coalescing operator

* Use nullish coalescing operator

* Make "type" a writable property

* Fix leftover isDataset usage

* Use "type" instead of "dataset" in unit tests

* Use "type" instead of "dataset" when generating URLs

* Use "type" instead of "dataset" in unit tests

* Deprecate Card["dataset"]

* Revert "Use "type" instead of "dataset" in unit tests"

This reverts commit 4e044103118607b50f799e1d387bf694cee2a3ae.

* Revert "Use "type" instead of "dataset" when generating URLs"

This reverts commit e3bb570b919ec07288192cf3ae83c73af1254c0d.

* Revert "Use "type" instead of "dataset" in unit tests"

This reverts commit ec23a5115e5a75f4afb1f83e874b70b0cfebf179.

* Revert "Fix leftover isDataset usage"

This reverts commit 6e85b8c76b281a1c330899ea35a2f02078b62836.

* Revert "Remove Question.prototype.isDataset"

This reverts commit eb152ddcc3a339b94005d69980b9321de8bdca70.

* Revert "Use Question.prototype.type instead of Question.prototype.isDataset"

This reverts commit 229ef6ca7ca4f5c8c37563614638e210c4f97b22.

* Deprecate Question.prototype.isDataset

* Make Question.prototype._setDataset private

* fix change turn to model does not change type

* Chris's suggestisons

* remove debug code

* Update 001_update_migrations.yaml

* s/datset/dataset

* Fix building ad-hoc questions based on models

* unparlalle test

* migration indents

* revision migration optimize for postgres and mysql

* Cal's suggestions

* grammar

* move type and dataset check to model

* use ensure-type-and-dataset-are-consistent on create-card

* Add "metric" CardType

* Mock create & update question endpoints

* Mock get & list question endpoints

* Fix casing

* Disable metrics mocks in tests to make CI green

* Disable mocks when cypress is active and in prod build

---------

Co-authored-by: Ngoc Khuat <qn.khuat@gmail.com>

kamilmielnik committed:
Merge branch 'master' into metrics-v2

nemanjaglumac committed:
[Metrics v2] Add "Metric" to the new entity menu (#38404)

* Add new `metric` item to the menu

* Add dummy route `/metric/query`

* Open new metric in a notebook mode

* Initialize new metric URL programatically using the existing helper

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Partially migrate is Question.prototype.isDataset (#38544)

* Migrate Question.prototype.isDataset in hasQuestionCacheSection

* Migrate Question.prototype.isDataset in metabase-enterprise/moderation

* Migrate Question.prototype.isDataset in ModelDetailPage

* Migrate Question.prototype.isDataset in navigateToNewCardFromDashboard

* Migrate Question.prototype.isDataset in handleQBInit

* Migrate Question.prototype.isDataset in Visualization

* Migrate Question.prototype.isDataset in QueryModals

* Migrate Question.prototype.isDataset in QuestionPane

* Migrate Question.prototype.isDataset in ObjectDetailView

* Migrate Question.prototype.isDataset in ArchiveQuestionModal

* Migrate Question.prototype.isDataset in Question.prototype.composeDataset

* Migrate Question.prototype.isDataset in getNativeQueryTable

* Migrate Question.prototype.isDataset in ViewFooter

* Convert SavedQuestionIntroModal to TS

* Migrate Question.prototype.isDataset in SavedQuestionIntroModal

* Migrate Question.prototype.isDataset in QuestionMoveToast

* Migrate Question.prototype.isDataset in QuestionInfoSidebar

* Fix QuestionMoveToast usage

* Fix unit tests

* Fix unit tests

* Make e2e test less flaky

* Make code work exactly like before

* Revert breaking change

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Pretend any question that includes "metric" in its name is a metric

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Remove question type toggling (#38654)

* Remove question type toggling
- it should no longer be required
- see @https://metaboat.slack.com/archives/C505ZNNH4/p1707400291390129

* Remove outdated comment

* Update unit tests

kamilmielnik committed:
New Metric page (`/metric/query`) (#38512)

* Add new `metric` item to the menu

* Add dummy route `/metric/query`

* Open new metric in a notebook mode

* Initialize new metric URL programatically using the existing helper

* Use type attribute in Question.prototype._serializeForUrl

* Extract getMultiStepTitle and add support for metrics

* Add support for metrics in getSingleStepTitle

* Combine getSingleStepTitle & getMultiStepTitle into getLabels

* Move nameInputPlaceholder computation into getLabels

* Use card.type in Urls.question

* Update getIsSavedQuestionChanged to account for metrics

* Remove dead code branch

* Always pass cardType to Urls.newQuestion to prevent Question.prototype.type from returning undefined

* Default type to "question" in getLabels for extra-safety

* Reorder conditions

* Use nullish coalescing operator instead of a ternary expression

* Remove no-longer-needed fallback

* Set reasonable defaults in Question.create

* Use model's UI for metrics

* Make Urls.newQuestion safer

* Do not show app bar in metrics pages

* Disable metadata tab for metrics

* Hide tabs for metrics

* Rename dataset to question

* Rename dataset to question

* Account for metric-based ad-hoc questions in updateQuestion

* Account for metrics in View

* Migrate Question.prototype.isDataset in hasQuestionCacheSection

* Migrate Question.prototype.isDataset in metabase-enterprise/moderation

* Migrate Question.prototype.isDataset in ModelDetailPage

* Migrate Question.prototype.isDataset in navigateToNewCardFromDashboard

* Migrate Question.prototype.isDataset in handleQBInit

* Migrate Question.prototype.isDataset in Visualization

* Migrate Question.prototype.isDataset in QueryModals

* Migrate Question.prototype.isDataset in QuestionPane

* Migrate Question.prototype.isDataset in ObjectDetailView

* Migrate Question.prototype.isDataset in ArchiveQuestionModal

* Migrate Question.prototype.isDataset in Question.prototype.composeDataset

* Migrate Question.prototype.isDataset in getNativeQueryTable

* Migrate Question.prototype.isDataset in ViewFooter

* Convert SavedQuestionIntroModal to TS

* Migrate Question.prototype.isDataset in SavedQuestionIntroModal

* Migrate Question.prototype.isDataset in QuestionMoveToast

* Migrate Question.prototype.isDataset in QuestionInfoSidebar

* Fix QuestionMoveToast usage

* Fix unit tests

* Fix unit tests

* Make e2e test less flaky

* Make code work exactly like before

* Revert breaking change

* Migrate Question.prototype.isDataset in QueryModals

* Add routing to state typing

* Add getRouting selector

* Infer entity type from URL when converting question back

* Remove creationType for metrics

* Make getCardTypeFromLocation more strict

* Replace Question.prototype.isDataset usage with Question.prototype.type

* Add mocks for routing state

* Use Location in getCardTypeFromLocation

* Ensure initialState contains "routing" only when using routerReducer
- this avoids getting warning from redux saying that initial state contains an extraneous key

* Extract createMockLocation

* Remove getCardTypeFromLocation hack, use state instead

* Revert routing changes

* Reduce diff

* Update updateQuestion unit tests

* Update QuestionDataSource unit tests

* Update QueryBuilder.unsaved-changes-warning.unit.spec.tsx

* Update ActionCreatorModal unit tests

* Update SaveQuestionModal unit tests

* Update comment

Co-authored-by: Nemanja Glumac <31325167+nemanjaglumac@users.noreply.github.com>

* Remove question type toggling
- it should no longer be required
- see @https://metaboat.slack.com/archives/C505ZNNH4/p1707400291390129

* Remove outdated comment

* Update unit tests

* Update reference/utils unit tests

* Revert unnecessary changes

* Fix unit tests

---------

Co-authored-by: Nemanja <31325167+nemanjaglumac@users.noreply.github.com>

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

nemanjaglumac committed:
[Metrics v2] Add new `/metric/:id` and `metric/:id/notebook` pages (#38554)

* Add new route for `/metric/:id`

* Alter the CollectionBadge to handle metrics

* Rename the component to `HeaderCollectionBadge`

* Do not offer turning a metric into a model

* Add new route for `/metric/:id/notebook`

* Do not offer to show and convert to SQL for metrics

* Guard against `question.type()` potentially being undefined

* Adjust propTypes

* Construct new, ad-hoc query upon opening a metric

* Fix the url when switching to a notebook mode

* Fix metric url hash showing up when you run a query

* Use more explicit condition for the ViewHeader breadcrumbs render check

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
FE - Unsaved Changes Warning for Metrics (#38666)

* Removed unused route

* Remove invalid test

* Remove invalid routes from isNavigationAllowed, update tests

* Remove invalid route

* Handle /model/:id/notebook route

* Migrate Question.prototype.isDataset to Question.prototype.type

* Consolidate code

* Add missing attribute

* Type cards and questions for extra safety

* Fix running edited models

* Improve diff

* Remove __support__/location

* Update getShouldShowUnsavedChangesWarning

* Update isNavigationAllowed with metrics

* Add tests for metrics

* Update test with metrics locations

* Fix isNavigationAllowed

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

nemanjaglumac committed:
[Metrics v2] Add the ability to edit a Metric definition page (#38664)

* Add new route for `/metric/:id/query`

* Expose "edit query" option in the menu

The "edit metadata" option stays hidden for now.

* Prevent fetching model indexes for metrics

* Account for the metrics when computing the dirty state

* Fix card type

* Account for potentially undefined `originalQuestion`

* Fix and expand the `DatasetEditor` unit tests

* Fix `QuestionActions` unit tests

* Fix the logic in the `getIsDirty` selector call stack

* Introduce `useModelIndexesListQuery` hook

* Use new `useModelIndexesListQuery` hook in `DatasetEditor`

* Fix type check failure

This commit remove the superfluous `query` prop from the `DatasetEditor`
component in the related unit test.

* Handle `modelIndexes` prop potentially being undefined

This commit fixes failing E2E tests due to the broken `fieldHasIndex` util.

kamilmielnik committed:
FE - Migrate card.dataset flag to card.type (#38714)

* Default to "question" in Question.prototype.type

* Replace "dataset" with "type"

* Remove Card['dataset'] attribute

* Replace "dataset" with "type"

* Replace "dataset" with "type"

* Remove redundant fallback values

* Remove Question.prototype._setDataset

* Replace "dataset" with "type"

* Replace "dataset" with "type"

* Replace "dataset" with "type"

* Remove redundant "dataset"

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Account for metrics

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Fix name after merge conflict

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Question.prototype.isDataset with Question.prototype.type

* Remove Question.prototype.isDataset

* Replace Question.prototype.isDataset with Question.prototype.type

* Replace Card['dataset'] with Card['type']

* Fix unit tests

* Fix test setup

* Make bookmark icons work again

* Fix updating questions

* Revert "Fix assertions"

This reverts commit d4b052e583e431c866ed6db300c7d415a159f3eb.

* Fix assertion

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

nemanjaglumac committed:
Add unit tests for `useModelIndexesListQuery` hook (#39145)

* Update the hook return type

* Add unit test

* Use model id from a const

* Add empty row between tests

metamben committed:
Enable saving metric type cards (#39262)

Co-authored-by: Kamil Mielnik <kamil@kamilmielnik.com>

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Fix post-merge leftover (automatic conflict resolution didn't get that part)

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

ranquild committed:
Merge master

ranquild committed:
merge master

metamben committed:
Merge remote-tracking branch 'origin/master' into metrics-v2

ranquild committed:
Pass cardType to canSave (#40519)

ranquild committed:
Fix merge

metamben committed:
Save metric v2 based query (#40411)

* Support filtering metric based queries

* Add metric aggregation when creating an ad-hoc query

* Fix test expecting the removed report_card.dataset field

* Comment out legacy metric tests

* Support filtering metric based queries

* Load metadata of the table underlying the metric

* Support saving metric based queries in MLv2 format

* Make metric based queries executable

* Support comparing queries

* Fix rebase

* Check metric validity on save

* Check if the card can be saved when updating

* Fix can-save tests

* Restrict can-save check in api/card to metrics

* Test that invalid metrics cannot be saved

* Fix some tests

* Let the FE normalize pMBQL queries too

* Fix question comparison

* Ignore e2e tests for old metrics

* Skip tests for legacy metrics

* Fix prettier

---------

Co-authored-by: Alexander Polyankin <alexander.polyankin@metabase.com>
Co-authored-by: Case Nelson <case@metabase.com>

snoe committed:
[Metrics V2] Allow metrics to be saved with breakouts (#41047)

A breakout on a metric will act as the default breakout when used by a
consuming query.

This tep allows saving the metric with a breakout.

snoe committed:
[Metrics V2] Create and execute queries based on metrics (#41099)

* [Metrics V2] Create and execute queries based on metrics

Fixes #41011
Fixes #41012
Fixes #41015

Updates query schema to use source-card and check metadata to create a
query based on a metric.

When a `:metadata/card` object is passed to `lib/query` such as through
`queryFromTableOrCardMetadata` the resulting query will have breakouts
copied from the source metric and a reference to the metric in the
aggregation.

* Update test/metabase/query_processor/middleware/metrics_test.clj

Co-authored-by: metamben <103100869+metamben@users.noreply.github.com>

* Update src/metabase/query_processor/middleware/metrics.clj

Co-authored-by: metamben <103100869+metamben@users.noreply.github.com>

* Update src/metabase/lib/query.cljc

Co-authored-by: metamben <103100869+metamben@users.noreply.github.com>

* Address PR review

---------

Co-authored-by: metamben <103100869+metamben@users.noreply.github.com>

metamben committed:
Fix visible-columns for metric-based queries (#41224)

Fixes #410130

metamben committed:
Add :metric? to cards' display-info (#41149)

Fixes #41014.

metamben committed:
Make sure aggregation references are resolvable (#41341)

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

kamilmielnik committed:
Merge branch 'master' into metrics-v2

snoe committed:
[Metrics v2] Search apis (#40932)

* [WIP] Search for metric report_cards

* Fix null value but present key

* Fix tests

* Update valid models for items endpoint

* Fix tests returning model literal

* Fix tests

* Fix search tests

* Drop legacy-metric revisions and related endpoints/test

* Fix search filter tests

* Drop tests related to legacymetric and revisions and related

* Fix dataset model change

* Add tests for metric related and revisions

* Update api changelog

* Keep related LegacyMetrics for xrays for now

snoe committed:
Merge remote-tracking branch 'origin/master' into metrics-v2

snoe committed:
Remove legacy metric permission test.

Now that metrics are under report_card the existing tests against
collections cover metrics. Data perms are no longer required to use
metrics.

metamben committed:
Handle joins in metric based queries (#41387)

Fixes #41200

kamilmielnik committed:
Merge branch 'master' into metrics-v2

metamben committed:
Fix tests in metrics v2 (#41601)

* Delete can-save native query zombies

* Make linter ignore skipped tests and disable a legacy metric test

kamilmielnik committed:
Merge branch 'master' into metrics-v2

ranquild committed:
Add metrics to DataSelector (#40442)

* Add metrics to DataSelector

* [WIP] Search for metric report_cards

* Fix null value but present key

* Fix tests

* FE fixes

* Update valid models for items endpoint

* Fix tests returning model literal

* Fix tests

* Fix search tests

* Drop legacy-metric revisions and related endpoints/test

* Fix search filter tests

* Drop tests related to legacymetric and revisions and related

* Fix dataset model change

* Add tests for metric related and revisions

* Update api changelog

* Keep related LegacyMetrics for xrays for now

* Use proper table id when using metric as data source

* Add metric to propTypes in SearchResults

* Add metrics support to DataPickerContainer

* Revert "Add metrics support to DataPickerContainer"

This reverts commit 7c6570058bcee9610298384ac397a0d7f8c4206d.

* Fixed types

* Fix types

* Fix opening a metric when clicking a search result

* Fix metrics loading

* Fix unit test

* ESLint

---------

Co-authored-by: Case Nelson <case@metabase.com>
Co-authored-by: Kamil Mielnik <kamil@kamilmielnik.com>

metamben committed:
Make dependent metadata look into metric definitions (#41655)

metamben committed:
Support creating questions with joined metrics (#41705)

* Replace legacy-metric usage with metric in MLv2

* Support creating questions with joined metrics

* Rename LegacyMetric... types to Metric...

metamben committed:
Merge master to metrics v2 (#41738)

* Add a way to update filter type and operator from parameter sidebar (#41555)

* Convert EmbeddingContext to Sdk Store slice (#41585)

* Add prefix to CSS custom properties (#41574)

* Migrate --border-*

* Migrate *breadcrumb*

* Migrate --color-bg-black

* Migrate --color-bg-dark

* Migrate --color-bg-light

* Migrate --color-bg-medium

* Migrate --color-bg-white

* Migrate --color-border

* Migrate --color-brand*

* Migrate --color-error

* Migrate --color-focus

* Migrate --color-shadow

* Migrate --color-success

* Migrate --color-text-dark

* Migrate --color-text-light

* Migrate --color-text-medium

* Migrate --color-text-white

* Migrate --color-white

* Migrate --default-border-radius

* Migrate --default-button-border-radius

* Migrate --default-font-family

* Migrate --default-font-size

* Migrate --default-header-margin

* Migrate --gap-1

* Migrate --icon-width

* Migrate --input-border-color

* Migrate --input-border-radius

* Migrate --margin-*

* Migrate --muted-color

* Migrate --padding-*

* Migrate --page-header-padding

* Migrate --subtitle-color

* Migrate --title-color

* Reset to master

* Remove unused --color-text-default

* Remove unused --color*

* Remove unused CSS custom properties

* Prefix only necessary CSS custom properties

* Fix Dispatch type (#41631)

* Fix Dispatch type

* Fix typing

* Fix types

* [Feature branch] Split data access and query builder access (#41581)

Co-authored-by: Noah Moss <noahbmoss@gmail.com>
Co-authored-by: Noah Moss <32746338+noahmoss@users.noreply.github.com>
Co-authored-by: Nick Fitzpatrick <nick@metabase.com>
Co-authored-by: John Swanson <john.swanson@metabase.com>
Co-authored-by: Sloan Sparger <sloansparger@users.noreply.github.com>
Co-authored-by: Sloan Sparger <sloansparger@gmail.com>

* Re-wire the `Groups` entity to use RTK Query under the hood (#41480)

* Delete unused endpoint from `PermissionsApi`

* Add group tag helpers

* Define RTK Query 5 main endpoints for `permissionApi` (group)

* Re-wire 5 main api methods in groups entity to work with RTK Query

* Re-wire `clearMember` action to work with RTK Query

* Add `Groups.api.update`

* Update types

* Fix membership failing to update after `CREATE_MEMBERSHIP`

* Make sure `members` is not undefined

* Fix cache invalidation

* Use dbname from database details instead of stats returned from Mongo (#41549)

* Use dbname from details instead of from stats

* Add mongo sharded cluster CI job

* add withUndos option to renderWithProviders to render Undos during … (#41576)

* Account for non-hydrated properties in GET /api/dashboard (#41622)

* Account for non-hydrated properties in GET /api/dashboard

* Fix provideTags

---------

Co-authored-by: Nemanja <31325167+nemanjaglumac@users.noreply.github.com>

* [docs] Revise Okta SAML docs (#41493)

* okta docs

* clarifying group mapping methods

* links

* Apply suggestions from code review

Co-authored-by: Jeff Bruemmer <jeff.bruemmer@gmail.com>

* review feedback

* Typo

---------

Co-authored-by: Jeff Bruemmer <jeff.bruemmer@gmail.com>

* Remove defaultauthdb portion of mongo connection string (#41553)

* Reset mapping after change of operator for native questions (#41615)

* Reset mapping after change of operator for native questions

* Add a basic test

* Better name for a test

* Add a new migration to clean up deprecated perm types (#41649)

* [QP, lib] Add new expression functions for host, domain, subdomain (#41540)

These functions are implemented with hairy regular expressions, and
it's more user-friendly and future-proof to name those functions in MBQL
rather than baking the `regexextract` and regex into the user's query.

It lets us evolve the regexes in the future if we detect a bug, and it
improves the UX since the user sees a meaningful function instead of

    regexextract([My URL Column], "(?<=[@\.])(?!www\.)[^@\.]+(?=\.[^@\.]{1,3}\.[^@\.]+$|\.[^@\.]+$)")

Also refactors the regexes somewhat so that they work for emails as well as URLs, and there's always just one layer of `:regex-match-first`.
Previously this was separated into two steps: URL or email to host, and host to (sub)domain.

Part of the follow-up for Extract Column epic #38964.

* New default DB permissions (#40869)

* Re-wire the `Revisions` entity to use RTK Query under the hood (#41558)

* Add RTK Query `revision` API

* Re-wire `Revisions` entity `list` to work with RTK Query

* Re-wire the `Revisions` entity `revert` to work with RTK Query

* Rename 'mac' alias to 'macaw' (#41680)

* Fix not sending necessary HTTP headers when downloading results (#41633)

* Fix not sending necessary HTTP headers when downloading results

* Fix fail E2E tests

* Add E2E test

* Fix failed tests

* Fix unit tests

* Fix download on subpath

* Fix unit tests for subpath

* Re-wire the `Snippets` entity to use RTK Query under the hood (#41656)

* Throw for `Snippets.api.delete`

* Add boilerplate `snippet` RTK API

* Add boilerplate RTK-compatibility layer for main `Snippets.api` endpoints

* Add types

* Add cache invalidation

* add validated migration checksums 45 46 47 and 48 (#41662)

* Modify JWT session token logic to use immediate return value from action (#41694)

* Port six collection-related components to TypeScript (#41632)

BaseItemsTable
BaseTableItem
BulkActions
CollectionContentView
CollectionCopyEntityModal
ItemsTable

* [tech debt] Stop using ms/BooleanString and only use ms/BooleanValue (#41390)

* [tech debt] Stop using ms/BooleanString and only use ms/BooleanValue

ms/BooleanString will coerce to a string, and we should probably get rid of it and only ever use ms/BooleanValue which will always coerce to a boolean
We want the coercion layer to give us values that are helpful, and true is more helpful than "true".
So we should remove ms/BooleanString and calls to parseBoolean

* Remove ms/BooleanString usage in api/native_query_snippet.clj

* Remove test case related to ms/BooleanString

* Remove the BooleanString schema def as no longer used

* Dates Binned by Week Export Formatting Matches App (#41619)

* Dates Binned by Week Export Formatting Matches App

Fixes: #41492

Before, the dates binned by week would be formatted as follows in CSV and JSON exports:

`Week 1 - 2024`

But that doesn't match the App's format of a Date Range:

`January 1, 2024 - January 7, 2024`

So, now the exports will apply formatting in this same way.

Note that a second bug (#41616) exists on the Frontend preventing the column formatting being applied in the app. This is
frontend only, and the csv and json exports will match the column formatting settings (eg. abbreviated dates and
alternative separators will be used in the export).

* add test and change default week style

* Week format test change

* also fix incorrect :hour formatting

* New default permissions for groups (#41323)

* Add dynamic fonts to embedding SDK (#41179)

* Fix TippyPopover default text color not visible after migration (#41716)

* Use the proper case for a "year" token format (#41715)

* Use proper case for year date format

Fixes #40493

* Add regression test for specific filters in different locales

* Actually use the locale

* Note the format_rows changes in the api changelog (#41268)

* Improve test coverage for describe-fks and describe-fields (#41528)

* Exclude internal content from admin checklist and usage stats (#41697)

* Replace `isLoggedIn` and `isInitialized` with `LoginStatus` (#41637)

* move cache business logic to model from api (#41677)

* Migrate Entity Copy Modal away from EntityForm (#41534)

* CopyQuestionForm

* CopyDashboardForm

* inital values adjustment, e2e tests

* FormObserver

* Remove unused `monospaceText` prop (#41721)

* Remove unused function (#41720)

* Be more consistent with borders and avoid double borders (#41565)

* Be more consistent with borders and avoid double borders

* Mark the aggregation 'Custom Expression' button as an action

* Remove color from action headers

* Remove bottom border from last child

* Remove top border from first child

* Make section header classes consistent

* Add border to search header

* Use correct color for borders

* Add props to enable borders in AccordionList

* Add borders to column picker elements

* dont use buildjet for e2e or frontend tests (#41733)

* dont use buildjet for e2e

* dont use buildjet for frontend either

---------

Co-authored-by: Uladzimir Havenchyk <125459446+uladzimirdev@users.noreply.github.com>
Co-authored-by: Oisin Coveney <oisin@metabase.com>
Co-authored-by: Mahatthana (Kelvin) Nomsawadi <me@bboykelvin.dev>
Co-authored-by: Kamil Mielnik <kamil@kamilmielnik.com>
Co-authored-by: Noah Moss <32746338+noahmoss@users.noreply.github.com>
Co-authored-by: Noah Moss <noahbmoss@gmail.com>
Co-authored-by: Nick Fitzpatrick <nick@metabase.com>
Co-authored-by: John Swanson <john.swanson@metabase.com>
Co-authored-by: Sloan Sparger <sloansparger@users.noreply.github.com>
Co-authored-by: Sloan Sparger <sloansparger@gmail.com>
Co-authored-by: Nemanja Glumac <31325167+nemanjaglumac@users.noreply.github.com>
Co-authored-by: lbrdnk <lbrdnk@users.noreply.github.com>
Co-authored-by: Nicolò Pretto <info@npretto.com>
Co-authored-by: Alex Yarosh <alexandra@metabase.com>
Co-authored-by: Jeff Bruemmer <jeff.bruemmer@gmail.com>
Co-authored-by: Braden Shepherdson <braden@metabase.com>
Co-authored-by: Tim Macdonald <tim@metabase.com>
Co-authored-by: bryan <bryan.maass@gmail.com>
Co-authored-by: Raphael Krut-Landau <raphael.kl@gmail.com>
Co-authored-by: A. Marius Rabenarivo <mariusrabenarivo@gmail.com>
Co-authored-by: adam-james <21064735+adam-james-v@users.noreply.github.com>
Co-authored-by: Cal Herries <39073188+calherries@users.noreply.github.com>
Co-authored-by: Alexander Solovyov <alexander@solovyov.net>
Co-authored-by: Romeo Van Snick <romeo@romeovansnick.be>
Co-authored-by: Ryan Laurie <30528226+iethree@users.noreply.github.com>

snoe committed:
[Metrics v2] process queries joining metrics (#41736)

* Replace legacy-metric usage with metric in MLv2

* Support creating questions with joined metrics

* Rename LegacyMetric... types to Metric...

* [Metrics v2] process queries joining metrics

* Update metrics expand to occur after fetch-source-query and rely on that to fill in source stages

* Change metrics/expand to leave stages in place rather than combining and to handle mixed source-card queries better

* Undo privacy change

* Add logging when metric ref cannot be replaced

* Comment out flaky test

* Comment out flaky test

---------

Co-authored-by: Tamás Benkő <tamas@metabase.com>

metamben committed:
Port backend v1 metric tests (#41808)

* Port underlying-records metric tests

* Port druid metrics test

* Port test-case-aggregation-in-metric

* Preserve aggregation name

* Port expression aggregation metric test

* Port share metric test

* Port sum-where metric test

* Port count-where metric test

* Port nested query metric test

* Simplify queries using mt/mbql-query

snoe committed:
[Metrics v2] join tests (#41853)

* [Metrics v2] Joins add tests and fixes

* Fix _id being referenced by snake-hating-map

* Throw on leftover metrics

* Remove legacy metric multimethods for :metric tag

* Disable expand-macro test

* Disable legacy metric tests

* Undo unnecessary change to returned-columns

snoe committed:
[Metrics v2] Drop stage metadata after manipulating metric stage #41918

* [Metrics v2] Drop stage metadata after manipulating metric stage

We can simply drop `lib/stage-metadata` from a metric stage after dropping the
aggregation. Not doing so, made QP assume that the aggregation was the
only field in that stage and make an explicit select for it.

* Add test assertion

snoe committed:
[Metrics V2] Fix model on activity endpoint (#41998)

Fixes #41993

ranquild committed:
Revert "Merge master to metrics v2 (#41738)"

This reverts commit 1a7e9865d970e3ed294a6a4190bb9b68d4094571.

ranquild committed:
Merge master

ranquild committed:
Check for canRun before showing the run button (#42000)

ranquild committed:
Use metric icon for search (#41995)

ranquild committed:
Use "Edit metric definition" for metrics (#41991)

ranquild committed:
Show metrics in collections (#41997)

ranquild committed:
Use default display for metrics (#42029)

metamben committed:
Port some metric v1 e2e tests (#42002)

* Port notebook e2e test
* Port nested metric e2e test
* Fix #41929.

metamben committed:
Enable metric preview only when an aggregation is selected (#42007)

Fixes #41928.

ranquild committed:
Add a tooltip for missing metric aggregations (#42041)

ranquild committed:
Fix metric icon (#42036)

ranquild committed:
FE metadata changes for metrics (#42106)

ranquild committed:
Fix data source test (#42156)

metamben committed:
Merge remote-tracking branch 'origin/master' into metrics-v2

ranquild committed:
Remove metrics v1 from admin (#42160)

ranquild committed:
Aggregation picker changes for metrics (#42053)

ranquild committed:
Fix editing an existing metric (#42231)

bshepherdson committed:
[MBQL lib] Add `metric-based?` function (#42221)

ranquild committed:
Hide "Basic metrics" when the query is entirely metric-based (#42233)

ranquild committed:
Open the expression editor if all metrics are selected (#42236)

metamben committed:
Port some metric v1 tests (#42125)

bshepherdson committed:
[MBQL lib] Suggest joining metrics on their primary breakout (#42222)

When joining two metrics `suggested-join-conditions` should return a condition
based on the primary breakouts of the two metrics, if their types are compatible.

Fixes #42155.

ranquild committed:
Metrics E2E tests (#42293)

ranquild committed:
Add metric icons in the notebook editor (#42368)

snoe committed:
[Metric V2] When joining a metric add its aggregation clause (#42300)

metamben committed:
Merge master

metamben committed:
Merge branch 'master' into metrics-v2-master-merge1

metamben committed:
Fix MetadataProviderable reference

metamben committed:
Fix MetricMetadata reference

ranquild committed:
Metrics E2E tests - part 2 (#42427)

ranquild committed:
Merge master

metamben committed:
Merge branch 'metrics-v2-master-merge1' into metrics-v2

metamben committed:
Silence the linter and disable the druid metrics test

metamben committed:
Comment out unused requires

ranquild committed:
Disable metric joins & revert join-related changes (#42477)

bshepherdson committed:
[Metrics] Remove support for joining on metrics (#42513)

Effectively a revert of #42222 and #42300.

Fixes #42471.

ranquild committed:
Add missing e2e tests for editing metrics (#42530)

metamben committed:
Migrate druid metrics-inside-aggregation-clauses-test to druid-jdbc

metamben committed:
Fix driver name

metamben committed:
Get a proof that the test runs

metamben committed:
Add more test debugging

metamben committed:
Add even more debugging

ranquild committed:
Metrics E2E tests part 3 (#42619)

camsaul commented on modules/drivers/druid-jdbc/test/metabase/driver/druid_jdbc/nested_field_columns_test.clj:
```diff
@@ -15,6 +15,7 @@
 (set! *warn-on-reflection* true)
 
 (deftest ^:parallel json-details-only-test
+  (is (= 0 1) "the nested test gets invoked")
```
shouldn't this fail? 😱

metamben replied:
Well spotted! I had doubts the tests actually run, so I added a couple of forced failures. As you see, they are not failing. Something to look into...

snoe committed:
[Metrics V2] Return metrics in table query-metadata (#42709)

Fixes #42668

Returns metric card definitions instead of legacy metrics
from /api/table/:id/query_metadata

ranquild committed:
Fix metric-based drills (#42727)

ranquild committed:
Allow hiding visualization for metrics in collections (#42728)

ranquild committed:
Allow replacing a card with a metric on a dashboard (#42732)

ranquild committed:
Properly pass compatible metrics to MBQL lib (#42762)

snoe committed:
[Metrics V2] Available metrics changes (#42720)

Fixes #42470

metamben committed:
Prevent druid from excluding druid-jdbc

metamben committed:
Remove forced failures and beautify test-runner

snoe committed:
[Metrics V2] Query processor handle no join strategy (#42666)

* [Metrics V2] Query processor handle no join strategy

Fixes #42462

Make QP preprocessor handle queries referencing metrics in the style of
legacy metrics. Enforce various compatibility checks.

* Make sure to adjust the actual stages being walked

* Address PR comments

* Add tests for recursion

* Fix e2e tests by setting fields on adjusted metric stage

* Fix more e2e tests by recurring properly

ranquild committed:
Merge master

ranquild committed:
Fix issues

ranquild committed:
Fix issues

ranquild committed:
Fix issues

ranquild committed:
Add metrics - hacky way

ranquild committed:
E2E fixes

ranquild committed:
Fix

ranquild committed:
E2E fixes

ranquild committed:
E2E fixes

ranquild committed:
Fix docstring and schema

ranquild committed:
E2E fixes

ranquild committed:
Fixes

ranquild committed:
Fix joins

ranquild committed:
Push changes

ranquild committed:
Push changes

ranquild committed:
Merge master once again

ranquild committed:
Fix test

metamben committed:
Exclude hidden tables from recent items

ranquild committed:
Fixes

ranquild committed:
Merge branch 'metrics-v2-master-may-17' of github.com:metabase/metabase into metrics-v2-master-may-17

ranquild committed:
Fix test

ranquild committed:
E2E fixes

metamben committed:
Merge branch 'metrics-v2' into migrate-druid-metric-aggregation-test

ranquild committed:
Fix tests

metamben committed:
Preserve FK key metadata on native models (#42866)

ranquild committed:
Merge branch 'metrics-v2' into migrate-druid-metric-aggregation-test

metamben committed:
Metric v2 migration (#42613)

ranquild committed:
Merge branch 'metrics-v2' into migrate-druid-metric-aggregation-test

lbrdnk committed:
Update legacy metrics test

lbrdnk committed:
Implement set-parameter and unprepare-value for LocalDateTime

lbrdnk committed:
Remove legacy metric test

metamben committed:
Merge remote-tracking branch 'origin/master' into migrate-druid-metric-aggregation-test

metamben committed:
Merge branch 'master' into migrate-druid-metric-aggregation-test

metamben committed:
Merge branch 'master' into migrate-druid-metric-aggregation-test
